src/L6/datasorter.py

```diff
@@ -18,7 +20,23 @@ def __init__(self):
 
     # Exercise 25
     def set_input_data(self, file_path_name: str):
-        pass
+        if type(file_path_name) is not str:
+            raise TypeError
+        file = Path(file_path_name)
+        if not file.is_file():
+            raise FileNotFoundError
+        elif not file_path_name.lower().endswith(".csv"):
+            raise NotCSVException
+        else:
+            with open(file_path_name, newline= '') as csvfile:
```

Comment:
Before you start populating `self.data` make sure to reset it to an empty list `self.data = []`. This will cover the case where, if the caller were to call `set_input_data` more than once, the list wouldn't contain old data.